src/GafferOSL/OSLObject.cpp

```diff
@@ -290,20 +295,27 @@ void OSLObject::compute( Gaffer::ValuePlug *output, const Gaffer::Context *conte
 			static_cast<StringPlug *>( output )->setToDefault();
 			return;
 		}
+		std::string primitiveVariablesToResample;
+
+		const OSLShader *shader = runTimeCast<const OSLShader>( shaderPlug()->source()->node() );
+		ConstShadingEnginePtr shadingEngine = shader ? shader->shadingEngine() : nullptr;
 
-		std::string nonConstantPrimitiveVariables;
+		auto ignoreVariable = [&shadingEngine](const std::string& name) -> bool
```

Comment:
I know we're feeling our way with C++11, but I don't think this is a good use for a lambda. One of the main benefits of lambdas is keeping logic close to the call site, but this is just moving it further away. Is there some other benefit I'm missing? If not, I suggest we just put this in the main loop below the original constancy test :

```
if( shadingEngine && !shadingEngine->needsAttribute( "", it->first ) )
{
    continue;
}
```

That's shorter, clearer, and fixes the potential bug I mentioned above...

Reply:
updated in 48476a7

Reply:
Thanks - merged.